### ISSUE TITLE
False-negative `unneeded-not` when using `hash` function

### ISSUE THREAD

**Issue by Moisan** [PyCQA/pylint#7688]
### Bug description

Line of the type `not hash(x) == hash(y)` does not trigger unneeded-not.

More complete example:
```python
def test_unneeded_not():
    first_string = '123'
    second_string = '456'
    assert not first_string == second_string
    assert not hash(first_string) == hash(second_string)
    assert not int(first_string) == int(second_string)
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: C0116: Missing function or method docstring (missing-function-docstring)
test.py:4:11: C0113: Consider changing "not first_string == second_string" to "first_string != second_string" (unneeded-not)
test.py:6:11: C0113: Consider changing "not int(first_string) == int(second_string)" to "int(first_string) != int(second_string)" (unneeded-not)
```

### Expected behavior

To trigger `not-unneeded` on the `assert not hash(first_string) == hash(second_string)` line.

### Pylint version

```shell
pylint 2.15.5
astroid 2.12.12
Python 3.9.13 | packaged by conda-forge | (main, May 27 2022, 17:01:00)
[Clang 13.0.1 ]
```

**Comment by clavedeluna** [comment]
This is inference-related. When the node is a call to `hash`, it seems to not be able to infer and [this tuple](https://github.com/PyCQA/pylint/blob/main/pylint/checkers/refactoring/not_checker.py#L67) is `(None, None)`

Wonder if passing `context` could help?

**Comment by mbyrnepr2** [comment]
Hi folks. I wanted to check a detail before opening a MR for this in astroid. Sorry for tagging @cdce8p @DanielNoord @jacobtylerwalls but I know you spend time in the astroid repo & don't want to lose out on your insights.

Would it be preferable to add an inference function, `infer_hash` which always returns `nodes.Const(0)` similar to the [infer_str](https://github.com/PyCQA/astroid/blob/main/astroid/brain/brain_builtin_inference.py#L792) function or would you prefer to try to do something like [infer_bool](https://github.com/PyCQA/astroid/blob/main/astroid/brain/brain_builtin_inference.py#L591) which would require defining a function on specific nodes like for example the [Dict node ](https://github.com/PyCQA/astroid/blob/7a0b5626ae7011c682a1fdee2362823cb36adc69/astroid/nodes/node_classes.py#L2310)?

**Comment by DanielNoord** [comment]
I think `infer_bool` makes more sense as it actually allows us to try and start inferring hashes instead of always returning the same.
